src/settings.rs

```diff
@@ -55,30 +75,53 @@ impl Settings {
     }
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ServerLimits {
+    pub max_post_bytes: u32,
+    pub max_post_records: u32,
+    pub max_record_payload_bytes: u32,
+    pub max_request_bytes: u32,
+    pub max_total_bytes: u32,
+    pub max_total_records: u32,
+}
+
+impl Default for ServerLimits {
+    fn default() -> ServerLimits {
+        ServerLimits {
+            max_post_bytes: DEFAULT_MAX_POST_BYTES,
+            max_post_records: DEFAULT_MAX_POST_RECORDS,
+            max_record_payload_bytes: DEFAULT_MAX_RECORD_PAYLOAD_BYTES,
+            max_request_bytes: DEFAULT_MAX_REQUEST_BYTES,
+            max_total_bytes: DEFAULT_MAX_TOTAL_BYTES,
+            max_total_records: DEFAULT_MAX_TOTAL_RECORDS,
+        }
+    }
+}
+
 #[derive(Debug)]
 pub struct Secrets {
     pub master_secret: Vec<u8>,
     pub signing_secret: [u8; 32],
 }
 
 impl Secrets {
-    pub fn new(master_secret: &str) -> Secrets {
+    pub fn new(master_secret: &str) -> Self {
```

Comment:
I only learned this for the first time when I reviewed one of those pull requests yesterday. Hence going back and applying it to the other code I wrote in here, even though it's not strictly related.